src/java/org/apache/cassandra/db/ConsistencyLevel.java

```diff
@@ -218,8 +219,7 @@ public void validateForRead() throws InvalidRequestException
 
     public void validateForWrite(String keyspaceName, QueryState queryState) throws InvalidRequestException
     {
-        if (SchemaConstants.isUserKeyspace(keyspaceName))
-            Guardrails.disallowedWriteConsistencies.ensureAllowed(this, queryState);
+        Guardrails.disallowedWriteConsistencies.ensureAllowed(this, queryState);
```

Comment:
the code path which starts here does not check for the cases where `QueryState` is for internal calls, which means that the Guardrail would be triggered in that case - is it correct (cc @JeremiahDJordan )

Reply:
it applies to other cases in this file as well

Reply:
Maybe I misunderstand, but I see:
`validateForWrite` -> `Guardrails.disallowedWriteConsistencies.ensureAllowed` -> `Guardrail.triggersOn` -> `Guardrail.enabled` which checks `(queryState == null || queryState.isOrdinaryUser())`

I think this should take into account internal calls, right?

Reply:
well, the user is considered to be ordinary when it is not null, not system and not super. When the authentication is enabled and we create a query state for internal calls, the user is anonymous (which is neither null, system or super)

Reply:
Ah, ok, so the check added in `QueryState.isOrdinaryUser` is not considering an anonymous (internal) user;
and this possibility happens when authentication is enabled.
So the questions are, why is this not being included and is that correct? Was it intentional in the original implementation?

Reply:
This is the same logic we have in DSE.  If we want to dig in more if that is correct lets open a new ticket.  But I think this is right for now.

Reply:
I will add a `!getClientState().isInternal` check in `QueryState.isOrdinaryUser`